platforms/jvm/plugins-java/src/integTest/groovy/org/gradle/api/plugins/DefaultJvmSoftwareComponentTest.groovy

```diff
@@ -16,195 +16,41 @@
-        (configuration.attributes.getAttribute(Bundling.BUNDLING_ATTRIBUTE) as Bundling).name == Bundling.EXTERNAL
-        (configuration.attributes.getAttribute(DocsType.DOCS_TYPE_ATTRIBUTE) as DocsType).name == DocsType.JAVADOC
-        project.tasks.getByName(component.mainFeature.sourceSet.javadocJarTaskName)
-        component.usages.find { it.name == JvmConstants.JAVADOC_ELEMENTS_CONFIGURATION_NAME}
+        expect:
+        project.objects.newInstance(DefaultJvmSoftwareComponent, "name")
+        project.objects.newInstance(DefaultJvmSoftwareComponent, "name2")
+        project.objects.newInstance(DefaultJvmSoftwareComponent, "name3")
     }
 
-    def "can configure sources jar variant"() {
-        when:
+    def "can add multiple feature instances to the same component"() {
+        given:
         project.plugins.apply(JavaBasePlugin)
 
-        // Verify the JavaBasePlugin does not create the below tested objects so we can
-        // ensure the DefaultJvmSoftwareComponent is actually creating these domain objects.
-        then:
-        project.configurations.findByName(JvmConstants.SOURCES_ELEMENTS_CONFIGURATION_NAME) == null
-        project.tasks.findByName("sources") == null
-
         when:
-        def component = project.objects.newInstance(DefaultJvmSoftwareComponent, "name", createFeature("main"), project.getExtensions())
-        component.withSourcesJar()
-
-        then:
-        def configuration = project.configurations.getByName(JvmConstants.SOURCES_ELEMENTS_CONFIGURATION_NAME)
-        configuration != null
-        !configuration.visible
-        !configuration.canBeResolved
-        configuration.canBeConsumed
-        (configuration.attributes.getAttribute(Usage.USAGE_ATTRIBUTE) as Usage).name == Usage.JAVA_RUNTIME
-        (configuration.attributes.getAttribute(Category.CATEGORY_ATTRIBUTE) as Category).name == Category.DOCUMENTATION
-        (configuration.attributes.getAttribute(Bundling.BUNDLING_ATTRIBUTE) as Bundling).name == Bundling.EXTERNAL
-        (configuration.attributes.getAttribute(DocsType.DOCS_TYPE_ATTRIBUTE) as DocsType).name == DocsType.SOURCES
-        project.tasks.getByName(component.mainFeature.sourceSet.sourcesJarTaskName)
-        component.usages.find { it.name == JvmConstants.SOURCES_ELEMENTS_CONFIGURATION_NAME}
-    }
-
-    def "has expected variants"() {
-        when:
-        project.plugins.apply(JavaBasePlugin)
-        def ext = project.getExtensions().getByType(JavaPluginExtension.class)
-        def component = project.objects.newInstance(DefaultJvmSoftwareComponent, "name", createFeature("main"), project.getExtensions())
+        def component = project.objects.newInstance(DefaultJvmSoftwareComponent, "name")
 
         then:
-        component.usages*.name == ["apiElements", "runtimeElements"]
+        component.features.add(createFeature("main"))
+        component.features.add(createFeature("foo"))
+        component.features.add(createFeature("bar"))
```

Comment:
❌ This really seems like it should be the `when` part of this test and the `then` part would be retrieving the component's features contains the created features.